composer/callbacks/system_metrics_monitor.py

```diff
@@ -46,9 +90,25 @@ def run_event(self, event: Event, state: State, logger: Logger):
         ]:
             local_node_system_metrics = self.compute_system_metrics()
             all_system_metrics = dist.all_gather_object(local_node_system_metrics)
-            system_metrics = {
-                key: value for local_metrics in all_system_metrics for key, value in local_metrics.items()
-            }
+            system_metrics = {}
+            gpu_metrics_set = set(_GPU_METRICS)
+
+            if self.log_all_data:
+                for rank, metrics in enumerate(all_system_metrics):
+                    for key, value in metrics.items():
+                        if key in gpu_metrics_set:
+                            system_metrics[f'Rank_{rank}_{key}'] = value
+                        else:
+                            system_metrics[key] = value
+
+            else:
+                model_device = next(state.model.parameters()).device
```

Comment:
should use `state.device` instead?

Reply:
I'm following memory_monitor's usage of reduce_value and unit testing, which uses the device of next(state.model.parameters()) - switching to state.device broke unit testing for me

Reply:
Discussed offline, sent API spec